examples/stm32/f4/stm32f4-discovery/usb_cdcacm/cdcacm.c

```diff
@@ -182,27 +190,28 @@ static int cdcacm_control_request(usbd_device *usbd_dev,
 		 * even though it's optional in the CDC spec, and we don't
 		 * advertise it in the ACM functional descriptor.
 		 */
-		return 1;
+		return USBD_REQ_HANDLED;
 		}
 	case USB_CDC_REQ_SET_LINE_CODING:
 		if (*len < sizeof(struct usb_cdc_line_coding)) {
-			return 0;
+			return USBD_REQ_NOTSUPP;
 		}
 
-		return 1;
+		return USBD_REQ_HANDLED;
 	}
-	return 0;
+	return USBD_REQ_NOTSUPP;
 }
 
 static void cdcacm_data_rx_cb(usbd_device *usbd_dev, uint8_t ep)
 {
 	(void)ep;
-
-	char buf[64];
-	int len = usbd_ep_read_packet(usbd_dev, 0x01, buf, 64);
+	
+	size_t len = usbd_ep_read_packet(usbd_dev, DATA_OUT_EP, data_buf,
+		sizeof(data_buf));
```

Comment:
I do not like large structures on stack in embedded projects because the stack tends to be tight. Not in this case, but people copy examples and mix them together with who knows what.